### DISCUSSION TITLE
[Question] x-for with anonymous items

### DISCUSSION THREAD

**Discussion by rogervila** [alpinejs/alpine#747] [4. General]
Hello,

I would like to create a list of anonymous items, but I am not sure how to do it with alpine.

I have tried a vue-like syntax like this one:

```html
<div x-data>
  <ul>
    <template x-for="i in 10">
      <li>test</li>
    </template>
  </ul>
</div>
```

I also tried to create an empty array but it did not work.

```html
<div x-data="{'items': new Array(10)}">
  <ul>
    <template x-for="item in items">
      <li>test</li>
    </template>
  </ul>
</div>
```

What would be the right way to achieve it?

Thank you.

**Comment by ryangjchandler** [comment]
@rogervila You need to use the following syntax: `[...Array(10).keys()]`, instead.

**Comment by rogervila** [comment]
@ryangjchandler thank you!

I have created a pen as an example: https://codepen.io/rogervila-1472551995/pen/KKzyQwZ
